app/controllers/application_controller/buttons.rb

```diff
@@ -676,10 +667,10 @@ def group_set_form_vars
         @edit[:new][:fields].push([mem.name,mem.id])
       end
     end
-    uri = CustomButton.buttons_for(@sb[:applies_to_class]).sort_by(&:name)
-    uri.each do |u|
-      @edit[:new][:available_fields].push([u.name,u.id]) if u.parent.nil?
-    end
+    @edit[:new][:available_fields] = CustomButton.buttons_for(@sb[:applies_to_class])
+                                                 .sort_by(&:name)
+                                                 .select  { |u| u.parent.nil? }
+                                                 .collect { |u| [u.name, u.id] }
```

Comment:
`parent` comes via `acts_as_miq_set_member` on `CustomButton` model.  Would need to dig in to  understand ActiveRecord implications.